score/ScoreBonus.py

```diff
@@ -0,0 +1,275 @@
+from nameinternal import get_card_metadata, get_score_bonus
+from collections import Counter, namedtuple
+from save import Savefile
+
+class ScoreBonus():
+    def __init__(self, name: str):
+        self.name = ""
+        self.format_string = ""
+        self.description = ""
+        self.score_bonus = 0
+
+        if name:
+            data = get_score_bonus(name)
+            self.name = data["NAME"]
+            if data["DESCRIPTIONS"]:
+                self.description = data["DESCRIPTIONS"][0]
+
+    def get_format_string(self) -> str:
+        if self.format_string:
+            return self.format_string
+        else:
+            return f'{self.name}: {self.score_bonus}'
+
+def get_ascension_score_bonus(save: Savefile) -> ScoreBonus:
+    bonus = ScoreBonus("Ascension")
+    if save.ascension_level > 0:
+        # Only applies to the score from the following bonuses:
+        # Floors Climbed
+        # Enemies Killed
+        # Elites Killed
+        # Bosses Slain
+        # Champion
+        # Perfect
+        # Beyond Perfect
+        # Overkill
+        # C-c-c-combo.
+        bonuses_total = 0
+        bonuses_total += get_floors_climbed_bonus(save).score_bonus
+        bonuses_total += get_enemies_killed_bonus(save).score_bonus
+        bonuses_total += get_act1_elites_killed_bonus(save).score_bonus
+        bonuses_total += get_act2_elites_killed_bonus(save).score_bonus
+        bonuses_total += get_act3_elites_killed_bonus(save).score_bonus
+        bonuses_total += get_bosses_slain_bonus(save).score_bonus
+        bonuses_total += get_champions_bonus(save).score_bonus
+        bonuses_total += get_perfect_bosses_bonus(save).score_bonus
+        bonuses_total += get_overkill_bonus(save).score_bonus
+        bonuses_total += get_combo_bonus(save).score_bonus
```

Comment:
This whole block would be better like `bonuses = sum([fn(save).score_bonus for fn in <insert list of all bonuses to count here>])`

This is also why I was advocating for having all of these in a dict (or otherwise accessible), then we can just grab said dict or list and use that.

Reply:
Wouldn't we still have to do a lookup for each one individually in a dict? We'll have to list them all out individually either way in some capacity

Reply:
We would, but I prefer to list them in a global dict than in the function. That also means we have only one place to maintain if we were to do similar stuff elsewhere.

Reply:
Could you show me how you would do it? I don't see how it'd actually be better to have it globally, in this scenario.

I think it could be helpful to put them all in a single list, then we could maybe share them between here at `Savefile._get_score_bonuses`, but this list specifically is not all of them. We could do a dictionary with perhaps the Score bonus name as the key, but I think it's also pretty odd to hardcode the keys we need for the Ascension bonus.